src/main/java/picard/illumina/CollectIlluminaLaneMetrics.java

```diff
@@ -176,29 +178,34 @@ public static void collectLaneMetrics(final File runDirectory, final File output
                                               final MetricsFile<MetricBase, Comparable<?>> laneMetricsFile,
                                               final MetricsFile<MetricBase, Comparable<?>> phasingMetricsFile,
                                               final ReadStructure readStructure, final String fileExtension,
-                                              final ValidationStringency validationStringency,
-                                              final boolean isNovaSeq) {
-            final Map<Integer, ? extends Collection<Tile>> laneTiles = readLaneTiles(runDirectory, readStructure, validationStringency, isNovaSeq);
+                                              final ValidationStringency validationStringency) {
+            int tileMetricsVersion = determineTileMetricsVersion(runDirectory, readStructure);
+            final Map<Integer, ? extends Collection<Tile>> laneTiles = readLaneTiles(runDirectory, readStructure, validationStringency, tileMetricsVersion);
             writeLaneMetrics(laneTiles, outputDirectory, outputPrefix, laneMetricsFile, fileExtension);
-            writePhasingMetrics(laneTiles, outputDirectory, outputPrefix, phasingMetricsFile, fileExtension, isNovaSeq);
+            writePhasingMetrics(laneTiles, outputDirectory, outputPrefix, phasingMetricsFile, fileExtension, tileMetricsVersion);
         }
 
-        public static File writePhasingMetrics(final Map<Integer, ? extends Collection<Tile>> laneTiles, final File outputDirectory,
+        private static int determineTileMetricsVersion(File illuminaRunDirectory, ReadStructure readStructure) {
+            final List<File> tileMetricsOutFiles = TileMetricsUtil.findTileMetricsFiles(illuminaRunDirectory, readStructure.totalCycles);
```

Comment:
Do you need to find ALL the tileMetricsFiles for all the cycles, since you are only retrieving element 0 below?   I'm assuming the version in element 0 will be the same as in all tileMetricsOutFiles?  (possible I am not understanding cycles vs. tiles here).

Reply:
Yeah I was leveraging a method that already exists to find the TileMetricsFiles even though I really only need to check one. I could run through them all and assert that they are all ==